ESTestCaseTests#testBasePortGradle failed

**Build scan**:
[scan](https://gradle-enterprise.elastic.co/s/4urzdm54n6kvq/tests/:test:framework:test/org.elasticsearch.test.test.ESTestCaseTests/testBasePortGradle#1) 

**Repro line**:
```
./gradlew ':test:framework:test' --tests "org.elasticsearch.test.test.ESTestCaseTests.testBasePortGradle" -Dtests.seed=92562F59F82C1285 -Dtests.security.manager=true -Dtests.locale=ga -Dtests.timezone=Asia/Makassar -Druntime.java=11
```

**Reproduces locally?**: Not for me!

**Applicable branches**: 7.x and master

**Failure history**:
[started June 13th](https://build-stats.elastic.co/goto/6ad9adfff13293df7437e1984f4b2635)

**Failure excerpt**:
```
java.lang.AssertionError: Values should be different. Actual: 10300
...
t org.elasticsearch.test.test.ESTestCaseTests.testBasePortGradle(ESTestCaseTests.java:195)
...
```

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -1409,10 +1409,17 @@ public abstract class ESTestCase extends LuceneTestCase {
         // Ephemeral ports on Linux start at 32768 so we modulo to make sure that we don't exceed that.
         // This is safe as long as we have fewer than 224 Gradle workers running in parallel
         // See also: https://github.com/elastic/elasticsearch/issues/44134
-        final String workerId = System.getProperty(ESTestCase.TEST_WORKER_SYS_PROPERTY);
-        // we adjust the gradle worker id with mod so as to not go over the ephemoral port ranges, but gradle continually
-        // increases this value, so the mod can eventually become zero, thus we shift on both sides by 1
-        final int startAt = workerId == null ? 0 : Math.floorMod(Long.valueOf(workerId) - 1, 223) + 1;
+        final String workerIdStr = System.getProperty(ESTestCase.TEST_WORKER_SYS_PROPERTY);
+        final int startAt;
+        if (workerIdStr == null) {
+            startAt = 0; // IDE
+        } else {
+            // we adjust the gradle worker id with mod so as to not go over the ephemoral port ranges, but gradle continually
+            // increases this value, so the mod can eventually become zero, thus we shift on both sides by 1
+            final long workerId = Long.valueOf(workerIdStr);
+            assert workerId >= 1 : "Non positive gradle worker id: " + workerIdStr;
+            startAt = Math.floorMod(workerId- 1, 223) + 1;
+        }
         assert startAt >= 0 : "Unexpected test worker Id, resulting port range would be negative";
         return 10300 + (startAt * 100);
     }

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -1410,7 +1410,9 @@ public abstract class ESTestCase extends LuceneTestCase {
         // This is safe as long as we have fewer than 224 Gradle workers running in parallel
         // See also: https://github.com/elastic/elasticsearch/issues/44134
         final String workerId = System.getProperty(ESTestCase.TEST_WORKER_SYS_PROPERTY);
-        final int startAt = workerId == null ? 0 : Math.floorMod(Long.valueOf(workerId), 223);
+        // we adjust the gradle worker id with mod so as to not go over the ephemoral port ranges, but gradle continually
+        // increases this value, so the mod can eventually become zero, thus we shift on both sides by 1
+        final int startAt = workerId == null ? 0 : Math.floorMod(Long.valueOf(workerId) - 1, 223) + 1;
         assert startAt >= 0 : "Unexpected test worker Id, resulting port range would be negative";
         return 10300 + (startAt * 100);
     }

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -1418,7 +1418,7 @@ public abstract class ESTestCase extends LuceneTestCase {
             // increases this value, so the mod can eventually become zero, thus we shift on both sides by 1
             final long workerId = Long.valueOf(workerIdStr);
             assert workerId >= 1 : "Non positive gradle worker id: " + workerIdStr;
-            startAt = Math.floorMod(workerId- 1, 223) + 1;
+            startAt = Math.floorMod(workerId - 1, 223) + 1;
         }
         assert startAt >= 0 : "Unexpected test worker Id, resulting port range would be negative";
         return 10300 + (startAt * 100);

--- a/test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java
+++ b/test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java
@@ -188,7 +188,7 @@ public class ESTestCaseTests extends ESTestCase {
 
         assertThat(ESTestCase.TEST_WORKER_VM_ID, not(equals(ESTestCase.DEFAULT_TEST_WORKER_ID)));
     }
-    
+
     public void testBasePortGradle() {
         assumeTrue("requires running tests with Gradle", System.getProperty("tests.gradle") != null);
         // Gradle worker IDs are 1 based

--- a/test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java
+++ b/test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java
@@ -188,8 +188,7 @@ public class ESTestCaseTests extends ESTestCase {
 
         assertThat(ESTestCase.TEST_WORKER_VM_ID, not(equals(ESTestCase.DEFAULT_TEST_WORKER_ID)));
     }
-
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/58279")
+    
     public void testBasePortGradle() {
         assumeTrue("requires running tests with Gradle", System.getProperty("tests.gradle") != null);
         // Gradle worker IDs are 1 based

